plugins/color_panel/linux/src/color_panel_plugin.cc

```diff
@@ -89,13 +90,37 @@ class ColorPanelPlugin::ColorPanel {
   GtkWidget *gtk_widget_;
 };
 
-ColorPanelPlugin::ColorPanelPlugin()
-    : JsonPlugin(kChannelName), color_panel_(nullptr) {}
+// static
+void ColorPanelPlugin::RegisterWithRegistrar(
+    flutter_desktop_embedding::PluginRegistrar *registrar) {
+  auto channel =
+      std::make_unique<flutter_desktop_embedding::MethodChannel<Json::Value>>(
+          registrar->messenger(), kChannelName,
+          &flutter_desktop_embedding::JsonMethodCodec::GetInstance());
+  auto *channel_pointer = channel.get();
+
+  std::unique_ptr<ColorPanelPlugin> plugin(
+      new ColorPanelPlugin(std::move(channel)));
```

Comment:
Would it be better to use make_unique here?

Reply:
Unfortunately make_unique doesn't work with classes whose constructors are private; see https://abseil.io/tips/134

The question is a good reminder to a comment as the tip suggests, so I've done that :)